mantidproject/mantid
# Refactor common code from SNSPowderReduction to utils

~~Do not merge before #30205~~

**Description of work.**

Pulls some absorption correction functions which calculate the absorption correction and create a donor workspace out of `SNSPowderReduction` and move them into a new file in the `mantid.utils` module so that they can be reused in other places.

This also adds a doc page for `mantid.utils` with a brief explanation and example codes. A testing directory is also made to run unit tests on code added to `mantid.utils`. 

**To test:**

Run the `SNSPowderRedux` system test to make sure the code refactor did not break the algorithm, and run the new unit tests with `ctest -R python.utils`.

*There is no associated issue.*


The version of this into `ornl-next` is #30194

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
